tests/cases/compiler/jsxElementType.tsx

```diff
@@ -0,0 +1,73 @@
+// @strict: true
+// @jsx: react
+/// <reference path="/.lib/react16.d.ts" />
+import * as React from "react";
+
+type React18ReactFragment = ReadonlyArray<React18ReactNode>;
+type React18ReactNode =
+  | React.ReactElement<any>
+  | string
+  | number
+  | React18ReactFragment
+  | React.ReactPortal
+  | boolean
+  | null
+  | undefined
+  | Promise<React18ReactNode>;
+
+// // React.JSXElementConstructor but it now can return React nodes from function components.
+type NewReactJSXElementConstructor<P> =
+  | ((props: P) => React18ReactNode)
+  | (new (props: P) => React.Component<P, any>);
+
+declare global {
+  namespace JSX {
+    type ElementType = string | NewReactJSXElementConstructor<any>;
+    interface IntrinsicElements {
+      ['my-web-component']: React.DOMAttributes<unknown>;
```

Comment:
nit - maybe call this `my-custom-element`

Reply:
done